src/containers/NamespacesDropdown/NamespacesDropdown.js

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { Dropdown } from 'carbon-components-react';
+
+import { fetchNamespaces } from '../../actions/namespaces';
+import { getNamespaces } from '../../reducers';
+
+class NamespacesDropdown extends React.Component {
+  componentDidMount() {
+    this.props.fetchNamespaces();
+  }
+
+  render() {
+    const { ...dropdownProps } = this.props;
+    return (
+      <div>
+        <Dropdown {...dropdownProps} />
+      </div>
+    );
+  }
+}
+
+NamespacesDropdown.defaultProps = {
+  label: 'Namespace',
+  items: ['loading...']
```

Comment:
I'd either leave this as an empty array or pre-populate with 'default' rather than 'loading...' since this value would actually be selected.

Alternatively, and probably better, expose the `isFetchingNamespaces` selector in reducers/index.js and use that to implement a real loading state using `DropdownSkeleton` or similar.

Reply:
Thanks Alan, I didn't know about the `DropdownSkeleton`. That is exactly what I wanted to do for loading, but didn't know what I could use to indicate the loading state 😄